pkg/api/api.go

```diff
@@ -219,3 +222,41 @@ func NewMessage(typ MessageType, pld json.RawMessage, parentID *ulid.ULID) *Mess
 		Payload:   pld,
 	}
 }
+
+func diff(old, new map[string]interface{}) map[string]interface{} {
+	r := make(map[string]interface{})
+	if reflect.DeepEqual(old, new) {
+		return nil
+	}
+	keys := make(map[string]struct{})
+	for k := range old {
+		keys[k] = struct{}{}
+	}
+	for k := range new {
+		keys[k] = struct{}{}
+	}
+	for k := range keys {
+		if reflect.DeepEqual(old[k], new[k]) { //if there is a difference in the turtle state
```

Comment:
This comment is not required and does not make too much sense, given that this function does not know anything about turtle states